### PR TITLE
Feat: stop sending external_id for env0_aws_credentials resource

### LINKED ISSUE #591: Stop sending `external_id` for `env0_aws_credentials` resource
Hey hey!

for security reasons, the external_id for `env0_aws_credentials` is now resolved on the backend, so there no reason to support it anymore.

Please remove it from the resource entirely

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
fixes #591 

### Solution
1. Removed external_id from all the locations I found,
2. Updated unit tests, examples, acceptance tests, and integration tests.

### CODE REVIEW

File: env0/resource_cost_credentials.go [hunk 1]
```diff
@@ -23,13 +23,6 @@ func resourceCostCredentials(providerName string) *schema.Resource {
 			ForceNew:    true,
 			Required:    true,
 		},
-		"external_id": {
```
Comment (TomerHeber) [inline env0/resource_cost_credentials.go:26]: Removed also from cost credentials...
I was guessing this also needs to be removed. I hope this was correct.
(Worst case, I'll add it back).

**Comment by yaronya** [comment]
@TomerHeber is this ready for review?

**Comment by avnerenv0** [comment]
@yaronya it doesn't matter.

@TomerHeber sorry but we've had some blowback from our customers regarding this sudden change.
We'll need to make the `external_id` optional as a first deprecation step - with some wording in the docs that "it will be removed entirely in the near future".
Can you do that here? Or open another PR if you prefer? 

Thanks

**Comment by TomerHeber** [comment]
> @yaronya it doesn't matter.
> 
> @TomerHeber sorry but we've had some blowback from our customers regarding this sudden change. We'll need to make the `external_id` optional as a first deprecation step - with some wording in the docs that "it will be removed entirely in the near future". Can you do that here? Or open another PR if you prefer?
> 
> Thanks

@yaronya @avnerenv0 - closing this PR (can re-open in the future).
I will add a deprecated message instead.

**Comment by avnerenv0** [comment]
deprecation period is over - we want this removed

**Comment by avnerenv0** [comment]
@TomerHeber the API is going to ignore external_id from now on and we wanna remove it from the provider - can you resolve the conflicts here please?

**Review by avnerenv0 — APPROVED** [review]
please resolve the conflicts - other than that - ready to merge 👍

**Comment by avnerenv0** [comment]
[Assume role - deprecate External ID entirely](https://trello.com/c/tzxU3XZg/8997-assume-role-deprecate-external-id-entirely)

**Pull request merged**